tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -548,11 +548,32 @@ func (iptConfigurator *IptablesConfigurator) run() {
 		// save packet mark set by envoy.filters.listener.original_src as connection mark
 		iptConfigurator.iptables.AppendRuleV4(constants.PREROUTING, constants.MANGLE,
 			"-p", constants.TCP, "-m", "mark", "--mark", iptConfigurator.cfg.InboundTProxyMark, "-j", "CONNMARK", "--save-mark")
+		// If the packet is already marked with 1337, then return. This is to prevent mark envoy --> app traffic again.
+		iptConfigurator.iptables.AppendRuleV4(constants.OUTPUT, constants.MANGLE,
+			"-p", constants.TCP, "-o", "lo", "-m", "mark", "--mark", iptConfigurator.cfg.InboundTProxyMark, "-j", constants.RETURN)
+		for _, uid := range split(iptConfigurator.cfg.ProxyUID) {
+			// mark outgoing packets from envoy to workload by pod ip
+			// app call VIP --> envoy outbound -(mark 1338)-> envoy inbound --> app
+			iptConfigurator.iptables.AppendRuleV4(constants.OUTPUT, constants.MANGLE,
+				"!", "-d", "127.0.0.1/32", "-p", constants.TCP, "-o", "lo", "-m", "owner", "--uid-owner", uid, "-j", constants.MARK, "--set-mark", "1338")
+		}
+		for _, gid := range split(iptConfigurator.cfg.ProxyGID) {
+			// mark outgoing packets from envoy to workload by pod ip
```

Comment:
Was the original call 
Curl my_podip like the pr title says or curl my_svc ip that happens to come back to self like the comment here?

Reply:
https://drive.google.com/file/d/1A6SLvwiRcEUp8WEHEQfjiDQpylhiSP5O/view?usp=sharing  can help you understand the case